control/mpc_follower/src/mpc_follower_core.cpp

```diff
@@ -91,60 +108,124 @@ MPCFollower::MPCFollower() : nh_(""), pnh_("~"), tf_listener_(tf_buffer_)
+    "input/reference_trajectory", rclcpp::QoS{1},
+    std::bind(&MPCFollower::callbackTrajectory, this, _1));
+  sub_current_vel_ = create_subscription<geometry_msgs::msg::TwistStamped>(
+    "input/current_velocity", rclcpp::QoS{1},
+    std::bind(&MPCFollower::callbackCurrentVelocity, this, _1));
+  sub_steering_ = create_subscription<autoware_vehicle_msgs::msg::Steering>(
+    "input/current_steering", rclcpp::QoS{1}, std::bind(&MPCFollower::callbackSteering, this, _1));
 
   /* wait to get vehicle position */
-  while (ros::ok()) {
-    try {
-      tf_buffer_.lookupTransform("map", "base_link", ros::Time::now(), ros::Duration(5.0));
-      break;
-    } catch (tf2::TransformException & ex) {
-      ROS_INFO("[mpc_follower] is waitting to get map to base_link transform. %s", ex.what());
-      continue;
+  {
+    auto cti = std::make_shared<tf2_ros::CreateTimerROS>(
+      this->get_node_base_interface(), this->get_node_timers_interface());
+    tf_buffer_.setCreateTimerInterface(cti);
+
+    while (rclcpp::ok()) {
+      static constexpr auto input = "map", output = "base_link";
+      try {
+        tf_buffer_.waitForTransform(
+          input, output, get_clock()->now(), rclcpp::Duration(5.0),
+          [this](const std::shared_future<geometry_msgs::msg::TransformStamped> & tf) {
+            tf.get();
+          });
+      } catch (tf2::TimeoutException & ex) {
+        RCLCPP_INFO(
+          get_logger(), "[mpc_follower] is waiting to get %s to %s transform. %s", input, output,
+          ex.what());
+        continue;
+      }
     }
   }
 
   /* for debug */
-  pub_debug_marker_ = pnh_.advertise<visualization_msgs::MarkerArray>("debug/markers", 10);
-  pub_debug_mpc_calc_time_ = pnh_.advertise<std_msgs::Float32>("debug/mpc_calc_time", 1);
-  pub_debug_values_ = pnh_.advertise<std_msgs::Float32MultiArray>("debug/debug_values", 1);
-
-  /* dynamic reconfigure */
-  dynamic_reconfigure::Server<mpc_follower::MPCFollowerConfig>::CallbackType dyncon_f =
-    boost::bind(&MPCFollower::dynamicRecofCallback, this, _1, _2);
-  dyncon_server_.setCallback(dyncon_f);
+  pub_debug_marker_ = create_publisher<visualization_msgs::msg::MarkerArray>("debug/markers", 10);
+  pub_debug_mpc_calc_time_ = create_publisher<std_msgs::msg::Float32>("debug/mpc_calc_time", 1);
+  pub_debug_values_ = create_publisher<std_msgs::msg::Float32MultiArray>("debug/debug_values", 1);
+
+  /* get parameter updates */
+  set_param_res_ =
+    this->add_on_set_parameters_callback(std::bind(&MPCFollower::paramCallback, this, _1));
+}
+
+namespace
+{
+template <typename T>
+void update_param(
+  const std::vector<rclcpp::Parameter> & parameters, const std::string & name, T & value)
+{
+  auto it = std::find_if(
+    parameters.cbegin(), parameters.cend(),
+    [&name](const rclcpp::Parameter & parameter) { return parameter.get_name() == name; });
+  if (it != parameters.cend()) {
+    value = it->template get_value<T>();
```

Comment:
@TakaHoribe this template method allows a generic implementation ando to avoid `as_double` etc.